testctrl/proto/test_sessions.proto

```diff
@@ -0,0 +1,178 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package proto;
+
+option go_package = "github.com/grpc/grpc/testctrl/genproto/testctrl/svc";
+
+import "google/protobuf/duration.proto";
+import "google/protobuf/timestamp.proto";
+import "google/longrunning/operations.proto";
+import "src/proto/grpc/testing/control.proto";
+
+service TestSessions {
+  // Starts a set of test scenarios.
+  //
+  // This is a long-running operation that is managed by the Operations service
+  // on the same server.  The long-running operation will assign a unique
+  // identifier to the session.
+  //
+  // The unique identifier can be used to poll for the session's status and
+  // results.  Streaming is not supported at present.
+  //
+  // It can also be used to cancel while in progress, but cancellations operate
+  // like killing a running process. If they occur while results are being
+  // reported, the persistent storage may receive some but not all of them.
+  //
+  // For the specification of a google.longrunning.Operation message, see the
+  // Long-running AIP at https://aip.dev/151.
+  rpc StartTestSession(StartTestSessionRequest) returns (google.longrunning.Operation) {
+    option (google.longrunning.operation_info) = {
+      response_type: "TestSessionResults"
+      metadata_type: "TestSessionMetadata"
+    };
+  }
+}
+
+message Component {
+  enum Kind {
+    // No kind was specified, so the service will attempt to infer one.
+    //
+    // This inference works as follows:
+    //   1. If this field belongs to the driver_container of the
+    //      StartTestSessionRequest, this kind is inferred to be a driver.
+    //   2. Otherwise, this field is inferred to be a client worker.
+    UNSPECIFIED = 0;
+
+    // Orchestrates the test amongst workers and reports results.
+    DRIVER = 1;
+
+    // Processes incoming requests from client containers.
+    SERVER = 2;
+
+    // Sends outgoing requests to a server container.
+    CLIENT = 3;
+  }
+
+  // The container image and a version as it appears in a docker registry.
+  //
+  // When using GCR, this will likely be a fully-qualified URL and either an
+  // explicit tag or sha256 hash. For example, a Java worker image may be
+  // similar to one of the following:
+  //
+  //   - gcr.io/grpc-testing/java_worker:v3.2.19
+  //   - gcr.io/grpc-testing/java_worker@sha256:82b6360a84c19f23ed7ee9...
+  //
+  // The "latest" tag is automatically set by the registry, and there
+  // are no guarantees that it will point to a specific image. It should be
+  // avoided.
+  string container = 1;  // [(google.api.field_behavior) = REQUIRED];
```

Comment:
I removed the `oneof` with tags and sha256. I feel like the docker name syntax may be sufficient enough. This reduces the complexity of the request object. Let me know if you like this or want me to return to the prior approach.